docs/dapp/opl/router-protocol/pingpong/deploy.md

```diff
@@ -0,0 +1,324 @@
+                emit AckFromDestination(requestId, ackMessage);
+            }
+        }
+        ```
+  </details>
+
+## Important highlights of the contract
+
+### IGateway
+
+    ```solidity
+    import "@routerprotocol/evm-gateway-contracts/contracts/IGateway.sol";
+    ```
+
+    Router Protocol has a library of `evm-gateway-contracs` to help writing cross-chain dapps. 
+    We import the `IGateway.sol` contract here to register the contract and later call `iSend()`.
+
+
+### getRequestMetaData()
+
+    ```solidity
+        function getRequestMetadata(
+        uint64 destGasLimit,
+        uint64 destGasPrice,
+        uint64 ackGasLimit,
+        uint64 ackGasPrice,
+        uint128 relayerFees,
+        uint8 ackType,
+        bool isReadCall,
+        string memory asmAddress
+    ) public pure returns (bytes memory) {
+        bytes memory requestMetadata = abi.encodePacked(
+        destGasLimit,
+        destGasPrice,
+        ackGasLimit,
+        ackGasPrice,
+        relayerFees,
+        ackType,
+        isReadCall,
+        asmAddress
+        );
+        return requestMetadata;
+    }
+    ```
+
+    To call later the `iPing()` function you need the `requestMetaData`. 
+    The function `getRequestMetaData()` helps you to create the bytes object for it.
+
+### iPing()
+
+  ```solidity
+  function iPing(
+    string calldata destChainId,
+    string calldata destinationContractAddress,
+    string calldata str,
+    bytes calldata requestMetadata
+  ) public payable {
+    currentRequestId++;
+
+    bytes memory packet = abi.encode(currentRequestId, str);
+    bytes memory requestPacket = abi.encode(destinationContractAddress, packet);
+    gatewayContract.iSend{ value: msg.value }(
+      1,
+      0,
+      string(""),
+      destChainId,
+      requestMetadata,
+      requestPacket
+    );
+    emit NewPing(currentRequestId);
+  }
+  ```
+
+  `iPing()` is the function you will call on the source chain, it calls `iSend()` 
+  on the gateway to initiate the cross-chain message. 
+  In a later section, when will make the cross-chain call, we will get in detail the call parameters.
+
+
+### iReceive()
+
+  ```solidity
+  function iReceive(
+    string memory, //requestSender,
+    bytes memory packet,
+    string memory srcChainId
+  ) external returns (uint64, string memory) {
+    require(msg.sender == address(gatewayContract), "only gateway");
+    (uint64 requestId, string memory sampleStr) = abi.decode(
+      packet,
+      (uint64, string)
+    );
+    if (
+      keccak256(abi.encodePacked(sampleStr)) == keccak256(abi.encodePacked(""))
+    ) {
+      revert CustomError("String should not be empty");
+    }
+    pingFromSource[srcChainId][requestId] = sampleStr;
+
+    emit PingFromSource(srcChainId, requestId, sampleStr);
+
+    return (requestId, sampleStr);
+  }
+  ```
+
+  `iReceive()` will be called from the Gateway on the destination chain.
+
+## Compile contract in Remix
```

Comment:
I would omit Remix-specific instructions here and just reference the relevant Remix chapter in the beginning (https://docs.oasis.io/dapp/emerald/writing-dapps-on-emerald#create-dapp-on-emerald-with-remix---ethereum-ide until we revamp it).

Reply:
Yes as general info people that never have used Remix a link is fine, on the other side I don't think omitting Remix specifics is good for now, as else people will compile/deploy contracts with 0.8.26.